### PR TITLE
Fail malleus if the participant fails

### PR DESCRIPTION
**Pull request by gpolitis**
I refactored the code so that we can pull any exceptions that happened from the participant thread, and fail the overall test when it finishes.

### CODE REVIEW

File: src/test/java/org/jitsi/meet/test/MalleusJitsificus.java [hunk 1]
```diff
@@ -170,69 +176,61 @@ public void testMain(
...
-                    {
-                        t.join();
-                    }
-                }
+                // TODO we should cancel the futures since test has failed
+                errors.add(new Exception("Failed to disrupt the bridges.", e));
             }
         }
 
-        int minFailureTolerance = Integer.MAX_VALUE;
```
Comment (gpolitis) [inline src/test/java/org/jitsi/meet/test/MalleusJitsificus.java:217]: The logic that checks if the participant failed "too much" was moved into the participant thread code and changed into throwing an exception (that is now handled).

File: src/test/java/org/jitsi/meet/test/MalleusJitsificus.java [hunk 2]
```diff
@@ -170,69 +176,61 @@ public void testMain(
...
+            catch (ExecutionException e)
+            {
+                errors.add(e.getCause());
             }
         }
 
-        if (disruptBridges && minFailureTolerance < 0)
+        if (!errors.isEmpty())
         {
-            throw new Exception("Minimum failure tolerance is less than 0");
```
Comment (gpolitis) [inline src/test/java/org/jitsi/meet/test/MalleusJitsificus.java:230]: There are timeout exceptions that were not covered by this check before.

File: src/test/java/org/jitsi/meet/test/MalleusJitsificus.java [hunk 3]
```diff
@@ -170,69 +176,61 @@ public void testMain(
...
+            throw new Exception("Failed with multiple errors. Throws the primary.", errors.get(0));
         }
     }
 
-    private class ParticipantThread
-        extends Thread
+    private class MalleusTask
```
Comment (gpolitis) [inline src/test/java/org/jitsi/meet/test/MalleusJitsificus.java:232]: Maybe this needs to be renamed?

File: src/test/java/org/jitsi/meet/test/MalleusJitsificus.java [hunk 4]
```diff
@@ -241,15 +239,15 @@ public void testMain(
...
         WebParticipant participant;
-        public int failureTolerance;
         private String bridge;
 
-        public ParticipantThread(
+        public MalleusTask(
             int i, JitsiMeetUrl url, long durationMs, long joinDelayMs,
             boolean muteVideo, boolean muteAudio, String region,
-            int initialFailureTolerance)
+            boolean enableFailureDetection)
```
Comment (gpolitis) [inline src/test/java/org/jitsi/meet/test/MalleusJitsificus.java:250]: The boolean that Jonathan had introduced in an earlier version of this class was good enough.

**Review by damencho — APPROVED** [review]

**Pull request merged**